### PR TITLE
Switch instrument search to use the POST api url.

### LINKED ISSUE #113: Instrument search broken
It looks like the old _mobile has removed its search capability, so any of the `client.search_for*`functions now return a 404 error.
```python
>>> client.search_for_instrument(InstrumentType.STOCK, 'SAAB')
...
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://www.avanza.se/_mobile/market/search/STOCK?query=SAAB&limit=10
```

### PR DESCRIPTION
**Pull request by tjader**
This should resolve #113.

### CODE REVIEW

**Comment by Qluxzz** [comment]
The return models of the old and new URL are not compatible so this is also a breaking change.

Running the test suite `python -m unittest` confirms this.
We'd need to update the different return models to match the new ones.

**Comment by tjader** [comment]
The search results all have the same output, so I merged them to use the same model. Also fixed a small issue in the transactions model (and added some bits into .gitignore that shouldn't be committed)

**Comment by Qluxzz** [comment]
LGTM, thanks for updating the tests and models!

File: avanza/models/search_result.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    INDEX = "INDEX"
+    PREMIUM_BOND = "PREMIUM_BOND"
+    SUBSCRIPTION_OPTION = "SUBSCRIPTION_OPTION"
+    EQUITY_LINKED_BOND = "EQUITY_LINKED_BOND"
+    CONVERTIBLE = "CONVERTIBLE"
+
+class SearchPrice(BaseModel):
+    # Most values are returned as strings.
+    # null values are returned as "null"
+    last: str|None
```
Comment (jonatanj) [inline avanza/models/search_result.py:25]: Note that this code makes the library incompatible with Python 3.9 (the minimum version stated in setup.py). See https://docs.python.org/3.10/whatsnew/3.10.html#pep-604-new-type-union-operator .

**Pull request merged**